modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseUpgradeMatcherReplacementCheck.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Nícolas Moura
+ */
+public abstract class BaseUpgradeMatcherReplacementCheck extends BaseFileCheck {
+
+	protected String afterFormat(
+		String fileName, String absolutePath, String content,
+		String newContent) {
+
+		return newContent;
+	}
+
+	protected String beforeFormat(
+		String fileName, String absolutePath, String content) {
+
+		return content;
+	}
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!isValidExtension(fileName)) {
+			return content;
+		}
+
+		String newContent = beforeFormat(fileName, absolutePath, content);
+
+		Matcher matcher = getPattern().matcher(content);
+
+		while (matcher.find()) {
+			newContent = format(content, newContent, matcher);
+		}
+
+		if (!content.equals(newContent)) {
+			newContent = afterFormat(
+				fileName, absolutePath, content, newContent);
+		}
+
+		return newContent;
+	}
+
+	protected abstract String format(
+		String content, String newContent, Matcher matcher);
+
+	protected abstract Pattern getPattern();
+
+	protected boolean isValidExtension(String fileName) {
+		if (!fileName.endsWith(".java")) {
+			return false;
+		}
+
+		return true;
+	}
```

Comment:
I think this should be changed an abstract method and have it receive a ___String array___ containing all valid extensions instead. Like `BaseJavaTermCheck.getCheckableJavaTermNames()`. That way, we do not end up with conditional statements like this:

```java
if (fileName.endsWith("java") || fileName.endsWith("jsp") || filename.endsWith("jspf") || filename.endsWith("vm")) {
    ...
}
```

Reply:
So just having checks provide something like this:

```java
return new String[] { "java", "jsp", "jspf", "vm" };
```